official/transformer/v2/transformer_main_test.py

```diff
@@ -42,49 +42,50 @@ class TransformerTaskTest(tf.test.TestCase):
 
   def setUp(self):
     temp_dir = self.get_temp_dir()
-    FLAGS.model_dir = temp_dir
-    FLAGS.init_logdir_timestamp = FIXED_TIMESTAMP
-    FLAGS.param_set = param_set = "tiny"
+    FLAGS.model_dir = os.path.join(temp_dir, FIXED_TIMESTAMP)
+    FLAGS.param_set = "tiny"
     FLAGS.use_synthetic_data = True
-    FLAGS.steps_per_epoch = 1
+    FLAGS.steps_between_evals = 1
+    FLAGS.train_steps = 2
     FLAGS.validation_steps = 1
-    FLAGS.train_epochs = 1
     FLAGS.batch_size = 8
-    FLAGS.init_weight_path = None
-    self.cur_log_dir = os.path.join(temp_dir, FIXED_TIMESTAMP)
-    self.vocab_file = os.path.join(self.cur_log_dir, "vocab")
-    self.vocab_size = misc.get_model_params(param_set, 0)["vocab_size"]
-    self.bleu_source = os.path.join(self.cur_log_dir, "bleu_source")
-    self.bleu_ref = os.path.join(self.cur_log_dir, "bleu_ref")
-    self.flags_file = os.path.join(self.cur_log_dir, "flags")
+    FLAGS.num_gpus = 1
+    FLAGS.distribution_strategy = "off"
+    self.model_dir = FLAGS.model_dir
+    self.temp_dir = temp_dir
+    self.vocab_file = os.path.join(temp_dir, "vocab")
+    self.vocab_size = misc.get_model_params(FLAGS.param_set, 0)["vocab_size"]
+    self.bleu_source = os.path.join(temp_dir, "bleu_source")
+    self.bleu_ref = os.path.join(temp_dir, "bleu_ref")
 
   def _assert_exists(self, filepath):
     self.assertTrue(os.path.exists(filepath))
 
   def test_train(self):
```

Comment:
#1 thank you for working on these tests.  

Question, do we need this test if we are not checking anything?  My assumption is we will catch exceptions via the other tests. Not remotely against the test just asking.

Reply:
Hm, these tests are basically verifying that we get to the end of train without running into any errors. 
I don't know of any other tests that verify this for transformer, are there any?